City-of-Helsinki/drupal-hdbt
# UHF-X prevent accordion error

## What was done
Instead of trying to guess sitename from url, pass the sitename from backend to frontend and read it from drupalSettings variable. 

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-X_prevent_accordion_error`
* Run `make drush-cr`

## How to test
- Go to a page with accordion or create one.
- (optional) Check with console that `window.drupalSettings.helfi_instance_name` variable exists.
- Go to check the localstorage and see that data with key {sitename}-accordion is updated when you operate the accordion

State won't work without sitename. You can simulate the situation.
- Go to hdbt.theme line 61, comment the line (the line where the $id is added to $variables array)
- Run `make drush-cr`
- Open the page incognito or make sure that  {sitename}-accordion local storage data does not exist.
- Go to check inspector. There should be no errors and accordion should work accordingly. State is not saved when operating accordion
